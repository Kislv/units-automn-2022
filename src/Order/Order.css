.Order {
    width: 60%;
    display: flex;
    flex-direction: column;
    align-items: flex-start;

    padding: 16px;
    border-radius: 12px;
}

.Order:hover {
    background-color: rgba(138, 137, 140, 0.1);
}

.Order-header {
    padding-bottom: 16px;
}

.Order-shop {
    font-weight: bold;
    margin-right: 16px;
    white-space: nowrap;
}

.Order-items {
    display: flex;
    flex-direction: column;
    flex-wrap: nowrap;
    align-items: flex-start;
}

.Order-item {
    padding: 8px 12px;
    border-bottom: 1px solid rgba(209, 209, 209, 0.24);
    white-space: nowrap;
}
