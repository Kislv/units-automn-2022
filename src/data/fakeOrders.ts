export type Order = {
	id?: number,
	date?: number,
	shop?: string,
	items?: string[],
}

export const fakeOrders: Order[] = [
	{
		id: 100,
		date: 1588359900000,
		shop: 'Сбереги Мега Маркер',
		items: [],
	},
	{
		id: 123,
		date: 1544356800000,
		shop: 'Alihandro Express',
		items: [
			'Утиный пластмасса для показ новый год',
			'Курица из нержавеющей стали, утка, гусь, голубь, питьевой фонтан',
			'Новый стиль один розница яйцо для упаковки форма латекс',
		]
	},
	{
		id: 124,
		date: 1652481120000,
		shop: 'Lamodник.ru',
		items: [
			'Жакет - BOREAL5',
			'Miss Gabby Костюм',
			'Ostin перчатки мужские',
			'Zara худи роз.',
		]
	},
	{
		id: 126,
		date: 1652585550000,
		shop: 'Эльдоградо',
		items: [
			'Ноутбук Apple MacBook Air 13.3" (MQD32RU/A)',
			'Игровая приставка Sony PlayStation 4 Pro 1TB Black (CUH-7208B)',
		]
	},
];
