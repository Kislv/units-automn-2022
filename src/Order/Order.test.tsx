import React from 'react';
import {OrderComponent} from './Order';
import Adapter from 'enzyme-adapter-react-16';
import {getDate} from '../utils/getDate';
import {shallow, configure} from 'enzyme';
jest.mock('../utils/getDate');
import { fakeOrders } from '../data/fakeOrders';

configure({ adapter: new Adapter() });

describe('Order component', () => {
	let wrapper;

	// Create mock before tests
	beforeEach(() => {
		(getDate as jest.Mock).mockReturnValue('2022-10-13');
	});

	// Clear mocks after tests
	afterEach(() => {
		jest.clearAllMocks();
	});


	it('order with undefined date', () => {
		const order = {
			shop: 'shop',
			date: undefined,
		};
		wrapper = shallow(<OrderComponent   order={order}/>);

		expect(wrapper.getElement()).toBeNull();
	});

	it('order with undefined shop', () => {
		const order = {
			shop: undefined,
			date: 5,
		};
		wrapper = shallow(<OrderComponent   order={order}/>);

		expect(wrapper.getElement()).toBeNull();
	});


	it('undefined order', () => {
		wrapper = shallow(<OrderComponent   order={undefined}/>);

		expect(wrapper.getElement()).toBeNull();
	});
	

	it('order with empty items', () => {
		wrapper = shallow(<OrderComponent order={fakeOrders[0]}/>);

		expect(wrapper).toMatchSnapshot();
		expect(getDate).toBeCalled();
	});

	it('render order with items', () => {
		wrapper = shallow(<OrderComponent order={fakeOrders[1]}/>);

		expect(wrapper).toMatchSnapshot();
		expect(getDate).toBeCalled();
	});

	it('order with undefined items', () => {
		const order = {
			shop: 'shop',
			date: 5,
			items: undefined,
		};
		wrapper = shallow(<OrderComponent   order={order}/>);

		expect(wrapper).toMatchSnapshot();
		expect(getDate).toBeCalled();
	});
});

