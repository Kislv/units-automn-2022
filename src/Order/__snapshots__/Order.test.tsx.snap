// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order component order with empty items 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      Сбереги Мега Маркер
    </span>
    <span
      id="date"
    >
      2022-10-13
    </span>
  </div>
  <div
    className="Order-items"
  >
    <div>
      Заказ пуст...
    </div>
  </div>
</div>
`;

exports[`Order component order with undefined items 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      shop
    </span>
    <span
      id="date"
    >
      2022-10-13
    </span>
  </div>
  <div
    className="Order-items"
  >
    <div>
      Заказ пуст...
    </div>
  </div>
</div>
`;

exports[`Order component render order with items 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      Alihandro Express
    </span>
    <span
      id="date"
    >
      2022-10-13
    </span>
  </div>
  <div
    className="Order-items"
  >
    <div
      className="Order-item"
      key="0"
    >
      Утиный пластмасса для показ новый год
    </div>
    <div
      className="Order-item"
      key="1"
    >
      Курица из нержавеющей стали, утка, гусь, голубь, питьевой фонтан
    </div>
    <div
      className="Order-item"
      key="2"
    >
      Новый стиль один розница яйцо для упаковки форма латекс
    </div>
  </div>
</div>
`;
